app/Http/Controllers/CompetitionsController.php

```diff
@@ -94,15 +95,38 @@ public function destroy(Competition $competition)
     }
 
     /**
-     * Download the CSV export of all users.
+     * Download a CSV export of all users based on different criteria.
      *
      * @param  \Gladiator\Models\Competition  $competition
-     * @return \League\Csv\ $csv
+     * @param  \Illuminate\Http\Request  $request
+     * @return void
      */
-    public function export(Competition $competition)
+    public function export(Competition $competition, Request $request)
     {
-        $csv = $this->manager->exportCSV($competition);
-        $csv->output('competition' . $competition->id . '.csv');
+        $fileName = 'contest_' . $competition->contest_id . '-' . 'competition_' . $competition->id;
```

Comment:
Potentially, might just be a really long file name though. Also using this for flash caching so it helps to have a shorter, predictable name I can check the cache for. Maybe a later upgrade though, w/ a function to switch spaces to hyphens, etc?